libtextsecure/sendmessage.js

```diff
@@ -1172,14 +1194,14 @@ MessageSender.prototype = {
       textsecure.storage.user.getNumber();
     proto.group.admins = [primaryDeviceKey];
 
-    return this.makeAttachmentPointer(avatar).then(attachment => {
-      proto.group.avatar = attachment;
-      // TODO: re-enable this once we have attachments
-      proto.group.avatar = null;
-      return this.sendGroupProto(recipients, proto, Date.now(), options).then(
-        () => proto.group.id
-      );
-    });
+    const attachment = await this.makeAttachmentPointer(avatar);
+
+    proto.group.avatar = attachment;
+    // TODO: re-enable this once we have attachments
+    proto.group.avatar = null;
+    await this.sendGroupProto(recipients, proto, Date.now(), options);
+
+    return proto.group.id;
```

Comment:
you just don't like the `.then` syntax?

Reply:
Yep. For one, it makes it hard to see what is being returned.